sa/storage-authority.go

```diff
@@ -165,28 +168,57 @@ func (ssa *SQLStorageAuthority) GetAuthorization(id string) (authz core.Authoriz
 		}
 		authD := authObj.(*authzModel)
 		authz = authD.Authorization
+	}
 
-		err = tx.Commit()
+	var challObjs []challModel
+	_, err = tx.Select(&challObjs, "SELECT * FROM challenges WHERE authorizationID = :authID", map[string]interface{}{"authID": authz.ID})
+	if err != nil {
+		tx.Rollback()
 		return
 	}
-	authD := *authObj.(*pendingauthzModel)
-	authz = authD.Authorization
+	var challs []core.Challenge
+	for _, c := range challObjs {
+		chall, err := modelToChallenge(&c)
+		if err != nil {
+			tx.Rollback()
+			return core.Authorization{}, err
+		}
+		challs = append(challs, chall)
+	}
+	authz.Challenges = challs
 
 	err = tx.Commit()
 	return
 }
 
-// Get the valid authorization with biggest expire date for a given domain and registrationId
+// GetLatestValidAuthorization gets the valid authorization with biggest expire date for a given domain and registrationId
 func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationId int64, identifier core.AcmeIdentifier) (authz core.Authorization, err error) {
 	ident, err := json.Marshal(identifier)
 	if err != nil {
 		return
 	}
-	err = ssa.dbMap.SelectOne(&authz, "SELECT id, identifier, registrationID, status, expires, challenges, combinations "+
+	err = ssa.dbMap.SelectOne(&authz, "SELECT id, identifier, registrationID, status, expires, combinations "+
 		"FROM authz "+
 		"WHERE identifier = :identifier AND registrationID = :registrationId AND status = 'valid' "+
 		"ORDER BY expires DESC LIMIT 1",
 		map[string]interface{}{"identifier": string(ident), "registrationId": registrationId})
+	if err != nil {
+		return
+	}
+
+	var challObjs []challModel
+	_, err = ssa.dbMap.Select(&challObjs, "SELECT * FROM challenges WHERE authorizationID = :authID", map[string]interface{}{"authID": authz.ID})
```

Comment:
This query would need to be updated in the same way. But instead, to consolidate, I'd recommend changing the query above to find the id of the latest valid authorization, and then call GetAuthorization with that id.